src/Rotalia/APIBundle/Controller/TransfersController.php

```diff
@@ -0,0 +1,222 @@
+<?php
+
+namespace Rotalia\APIBundle\Controller;
+
+use Exception;
+use Rotalia\APIBundle\Form\TransferType;
+use Rotalia\InventoryBundle\Component\HttpFoundation\JSendResponse;
+use Rotalia\InventoryBundle\Form\FormErrorHelper;
+use Rotalia\InventoryBundle\Model\Transfer;
+use Rotalia\InventoryBundle\Model\TransferPeer;
+use Rotalia\InventoryBundle\Model\TransferQuery;
+use Nelmio\ApiDocBundle\Annotation\ApiDoc;
+use Rotalia\UserBundle\Model\MemberQuery;
+use Rotalia\UserBundle\Model\User;
+use Symfony\Component\HttpFoundation\Request;
+use DateTime;
+
+class TransfersController extends DefaultController
+{
+
+    /**
+     * Fetch list of transfers
+     *
+     * @param Request $request
+     * @ApiDoc(
+     *     resource = true,
+     *     statusCodes = {
+     *          200 = "Returned when successful",
+     *          400 = "Returned when dateFrom or dateUntil are not in valid date format or both conventId and memberId filters are null",
+     *          403 = "Returned when user has insufficient privileges",
+     *     },
+     *     description="Fetch transfers list",
+     *     section="Transfers",
+     *     filters={
+     *          {"name"="conventId","type"="int","description"="Fetch reports for another convent than member home convent"},
+     *          {"name"="memberId","dataType"="integer","description"="Member ID of user in interest"},
+     *          {"name"="dateFrom","type"="string","description":"datetime string"},
+     *          {"name"="dateUntil","type"="string","description":"datetime string"},
+     *          {"name"="limit","type"="int","description":"Limit number of transfers returned, default 5"},
+     *          {"name"="offset","type"="int","description":"Offset transfers returned for pagination, default 0"},
+     *     }
+     * )
+     *
+     * @param Request $request
+     * @return JSendResponse
+     */
+    public function listAction(Request $request)
+    {
+        $conventId = $request->get('conventId', null);
+        $memberId = $request->get('memberId', null);
+        $dateFrom = $request->get('dateFrom', null);
+        $dateUntil = $request->get('dateUntil', null);
+        $limit = $request->get('limit', 5);
+        $offset = $request->get('offset', 0);
+
+        $transferQuery = TransferQuery::create();
+
+        $memberConventId = $this->getMember()->getKoondisedId();
+
+        if ($conventId === null && $memberId === null && !$this->isGranted(User::ROLE_SUPER_ADMIN)) {
+            return JSendResponse::createFail(['Täpsustada konvent või kasutaja'], 400);
+        }
+
+        if ($memberId !== null) {
+
+            if ($memberId == $this->getMember()->getId()) {
+                // User requests its own transfers
+                $transferQuery->filterByMemberId($memberId);
+
+            } else {
+                return JSendResponse::createFail(['Pärida saab ainult enda ülekandeid'], 400);
```

Comment:
Mul hetkel ei ole sellist use case - admin teeb päringuid konvnedi kaupa ning siis kasutaja kaupa saab kasutajaliidesest filtreerida.

Ei hakanud ilmaasjata keerulisemaks tegema _controllerit_, kui on soovi, siis ma võin selle ka muidugi lisada.